subprojects/docs/src/docs/userguide/dep-man/03-controlling-transitive-dependencies/platforms.adoc

```diff
@@ -42,6 +42,40 @@ include::sample[dir="snippets/dependencyManagement/catalogs-settings/groovy",fil
 include::sample[dir="snippets/dependencyManagement/catalogs-settings/kotlin",files="settings.gradle.kts[tags=simple_catalog]"]
 ====
 
+[NOTE]
+.Mapping of aliases to type-safe accessors
+====
+Aliases must consist of a series of identifiers separated by a dash (`-`, recommended), an underscore (`_`) or a dot (`.`).
+Identifiers themselves must consist of ascii characters, preferably lowercase, eventually followed by numbers.
+
+For example:
+
+- `guava` is a valid alias
+- `groovy-core` is a valid alias
+- `commons-lang3` is a valid alias
+- `androidx.awesome.lib` is also a valid alias
+- but `this.#is.not!`
+
+Then type safe accessors are generated for _each subgroup_.
+For example, given the following aliases:
+
+`guava`, `groovy-core`, `groovy-xml`, `groovy-json`, `androidx.awesome.lib`
+
+We would generate the following type-safe accessors:
+
+- `libs.guava`
+- `libs.groovy.core`
+- `libs.groovy.xml`
+- `libs.groovy.json`
+- `libs.androidx.awesome.lib`
+
+It is illegal to have an alias to a dependency which happens to also belong to a nested group.
+For example `commons` and `commons-core` is illegal.
```

Comment:
I have the impression that this is a sever limitation.
I understand the intention of splitting on `-`, `_` and `.` in the same way, but I wonder about the usability then.

I am also not sure if suggesting to split only on `.` but not on the others improves the situation and justifies the difference in behavior.

Wdyt?

Reply:
I would prefer to see automatic camel-casing of kebab-case group/artifact names when generating aliases. For example, `groovy-json`'s accessor is generated as `getGroovyJson()`.

Reply:
Understood. We have discussed why this is done this way. There are several reasons: 

- `-`, `.` and `_` can all be seen as non alphanumeric separators, which makes them behave consistently
- if you want camelCase accessors, you can declare the alias explicitly: `groovyJson = "2.4.5"` in which case there are _no_ sub-accessors generated: `libs.groovyJson`
- it is arguable whether you should use a sub-accessor for libraries having a common prefix. If you take `commons-lang3`, where we would, for this PR, generate `commons.lang3`, it makes sense to have a group because `commons` is actually a group of libraries of Apache (`commons-math3`, `commons-io`, ...) so all of them would be accessible under `commons.*` naturally. In case it doesn't make sense to create a group, you can use an explicit camel case alias.
- the last argument is that considering `.` only as a separator makes it way less convenient in the TOML file. If we decide to keep it, which is still under hot debates, then you'd have to write:

```
"kotlin.stdlib.jdk8" = 
"kotlin.stdlib.jdk7" = 
```
To actually generate `kotlin.stdlib.jdk7` accessor, because the key itself is dotted (so you need quotes for the alias). Of course we shouldn't take the decision only based on this, but it makes things significantly less elegant.